pyo3-macros-backend/src/pyclass.rs

```diff
@@ -2203,6 +2218,58 @@ fn pyclass_hash(
     }
 }
 
+fn pyclass_auto_new<'a>(
+    options: &PyClassPyO3Options,
+    cls: &syn::Ident,
+    fields: impl Iterator<Item = &'a &'a syn::Field>,
+    methods_type: PyClassMethodsType,
+    ctx: &Ctx,
+) -> Result<Option<syn::ItemImpl>> {
+    if options.auto_new.is_some() {
+        ensure_spanned!(
+            options.set_all.is_some(), options.hash.span() => "The `auto_new` option requires the `set_all` option.";
```

Comment:
Do we really need this restriction? It seems totally valid to just generate the constructor.